src/main/webapp/js/userMap.js

```diff
@@ -90,22 +104,106 @@ function handleData(err, userData) {
-                         + '<p>'
-                             + '<span class="bold">'
-                             + geo.properties.name
-                             + '</span>'
-                             + '<br>'
-                             + 'Institution(s): '
-                             + data.numOfInstitutions
-                         + '</p>'
-                     + '</div>';
+                return getTooltipContent({
+                    name: geo.properties.name,
+                    numOfInstitutions: data.numOfInstitutions
+                });
             }
         }
     });
+
+    map.addPlugin('pins', function (layer, data, options) {
+        var self = this,
+            fillData = this.options.fills,
+            svg = this.svg;
+
+        if (!data || (data && !data.slice)) {
+            handleError();
+            return;
+        }
+
+        var markers = layer.selectAll('image.datamaps-pins').data(data, JSON.stringify);
+
+        markers
+        .enter()
+        .append('image')
+        .attr('class', 'datamaps-pin')
+        .attr('xlink:href', 'images/pin.png')
+        .attr('height', 20)
+        .attr('width', 20)
+        .attr('x', getX)
+        .attr('y', getY)
+        .on('mouseover', function (datum) {
+            var $this = d3.select(this);
+
+            if (options.popupOnHover) {
+                self.updatePopup($this, datum, options, svg);
+            }
+        })
+        .on('mouseout', function (datum) {
+            var $this = d3.select(this);
+
+            if (options.highlightOnHover) {
+                var previousAttributes = JSON.parse($this.attr('data-previousAttributes'));
+                for (var attr in previousAttributes) {
+                    $this.style(attr, previousAttributes[attr]);
+                }
+            }
+            d3.selectAll('.datamaps-hoverover').style('display', 'none');
+        });
+
+        markers.exit()
+        .transition()
+        .delay(options.exitDelay)
+        .attr('height', 0)
+        .remove();
+
+        function getCoordinates(datum) {
+            var latLng;
+            if (datumHasCoords(datum)) {
+                latLng = self.latLngToXY(datum.latitude, datum.longitude);
+            } else if (datum.centered) {
+                latLng = self.path.centroid(svg.select('path.' + datum.centered).data()[0]);
+            }
+
+            return latLng;
+        }
+
+        function getX(datum) {
+            return getCoordinates(datum)[0];
+        }
+
+        function getY(datum) {
+            return getCoordinates(datum)[1];
+        }
+
+        function datumHasCoords (datum) {
+            return typeof datum !== 'undefined' && typeof datum.latitude !== 'undefined' && typeof datum.longitude !== 'undefined';
+        }
+    });
+
+    map.pins(pins, {
+        popupOnHover: true,
+        popupTemplate: getTooltipContent
+    });
+}
+
+function getTooltipContent(data) {
```

Comment:
Let's not pass the function more variable then it needs. Passing in two vars, `name` and `numOfInstitutions`, should be enough.